limitador/src/storage/redis/redis_async.rs

```diff
@@ -30,38 +32,51 @@ pub struct AsyncRedisStorage {
 
 #[async_trait]
 impl AsyncCounterStorage for AsyncRedisStorage {
+    #[tracing::instrument(skip_all)]
     async fn is_within_limits(&self, counter: &Counter, delta: i64) -> Result<bool, StorageErr> {
         let mut con = self.conn_manager.clone();
 
-        match con
-            .get::<String, Option<i64>>(key_for_counter(counter))
-            .await?
+        let span = trace_span!("datastore");
```

Comment:
I've named them the same (and open to changing the name if we want tho), so that they can be grouped in how we consume. I can see what time is spent talking to the "datastore". Hence the generic name as well, so that it can be named the same, whatever the actual underlying datastore is… If we'd want to add more details, a thing we've done elsewhere, is to add a field with the actual query sent to that datastore… 
Finally, tho afaik this might still not be possible with redis, the trace would propagate to the underlying datastore as well, so that you'd see yet more details as to what happens underneath that span.

Reply:
Cool yeah that sounds very reasonable, something we could look at down the line if we would like more info about the different datastore accesses.